hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -1746,6 +1785,10 @@ public int getAsyncClusterMaxCommits() {
     return getInt(HoodieClusteringConfig.ASYNC_CLUSTERING_MAX_COMMITS);
   }
 
+  public String getAvroPayloadClass() {
+    return getStringOpt(HoodiePayloadConfig.PAYLOAD_CLASS_NAME).orElseGet(() -> HoodieRecordPayload.getAvroPayloadForMergeMode(getRecordMergeMode()));
```

Comment:
I think it would be better to left existing Avro read path along without mixing the merge mode code in, i.e., Avro read path should completely operate on payload class only, without knowing anything about merge mode or so.  To make that work, the write config may need to automatically set `PAYLOAD_CLASS_NAME` if Avro record merger is used, to simplify the logic and make the code maintainable.

Reply:
Avro path is not going to be maintained as much as the newer paths. So this way makes the newer paths easier to maintain. Prioritizing the avro path over the default path isn't the way to go.

Reply:
We still need to main the Avro merger path for custom payload for some time.